Barotrauma/BarotraumaShared/SharedSource/LuaCs/Plugins/CsPackageManager.cs

```diff
@@ -0,0 +1,980 @@
+                }
+
+                if (packageChecksPredicate is not null && !packageChecksPredicate.Invoke(packageToProcess))
+                {
+                    unableToLoad.Add(packageToProcess, $"Unable to load package {packageToProcess.Name} due to failing checks.");
+                    completedPackages.Add(packageToProcess);
+                    return PackageProcRet.BadPackage;
+                }
+
+                currentNodeChain.Add(packageToProcess);
+
+                foreach (ContentPackage dependency in dependencies)
+                {
+                    // The mod lists a dependent that was not found during the discovery phase.
+                    if (!packages.ContainsKey(dependency))
+                    {
+                        // search to see if it's enabled
+                        if (!ContentPackageManager.EnabledPackages.All.Contains(dependency))
+                        {
+                            // present warning but allow loading anyways, better to let the user just disable the package if it's really an issue.
+                            ModUtils.Logging.PrintError(
+                                $"Warning: the ContentPackage of {packageToProcess.Name} requires the Dependency {dependency.Name} but this package wasn't found in the enabled mods list!");
+                        }
+
+                        continue;
+                    }
+
+                    var ret = ProcessPackage(dependency, packages[dependency]);
+
+                    if (ret is PackageProcRet.BadPackage)
+                    {
+                        if (!unableToLoad.ContainsKey(packageToProcess))
+                        {
+                            unableToLoad.Add(packageToProcess, $"Error: Dependency failure. Failed to load {dependency.Name}");
+                        }
+                        currentNodeChain.Remove(packageToProcess);
+                        if (!completedPackages.Contains(packageToProcess))
+                        {
+                            completedPackages.Add(packageToProcess);
+                        }
+                        return PackageProcRet.BadPackage;
+                    }
+                }
+                
+                currentNodeChain.Remove(packageToProcess);
+                completedPackages.Add(packageToProcess);
+                readyPackages.Add(packageToProcess); 
+                return PackageProcRet.Completed;
+            }
+        }
+        catch (Exception e)
+        {
+            ModUtils.Logging.PrintError($"Error while generating dependency loading order! Exception: {e.Message}");
+#if DEBUG
+            ModUtils.Logging.PrintError($"Stack Trace: {e.StackTrace}");
+#endif
+            cannotLoadPackages = unableToLoad.Any() ? unableToLoad : null;
+            return false;
+        }
+        cannotLoadPackages = unableToLoad.Any() ? unableToLoad : null;
+        return true;
+    }
+
+    private enum PackageProcRet : byte
```

Comment:
Does the underlying type (`byte`) actually matter? It's usually not necessary unless you're doing binary serialization.

Reply:
When I first planned this, I was originally going to network some things and this is a remnant of that.